crates/hdk/tests/integration.rs

```diff
@@ -0,0 +1,570 @@
+#![cfg(feature = "mock")]
+use std::ops::Range;
+
+use hdk::prelude::*;
+use holo_hash::hash_type::AnyLinkable;
+use Compare::*;
+
+mod integrity_zomes {
+    use holochain_deterministic_integrity::prelude::*;
+
+    #[hdk_entry_helper]
+    pub struct A;
+    #[hdk_entry_helper]
+    pub struct B;
+    #[hdk_entry_helper]
+    pub struct C;
+
+    pub mod integrity_a {
+
+        use super::*;
+
+        #[hdk_entry_defs(skip_hdk_extern = true)]
+        #[unit_enum(UnitFoo)]
+        pub enum EntryTypes {
```

Comment:
The convention should probably be to call this enum `EntryType` instead of `EntryTypes`. I know the enum defines all entry types, but an instance of the entry type represents one particular value of any one of the variants.

Reply:
We already have a type called `EntryType` in the integrity layer though. Also there's no restrictions on the name of this enum.

Reply:
Yeah but this is kind of setting the convention, most people are probably going to name this whatever we call it, precisely because the name is arbitrary. And enums are not usually pluralized. Can you think of anything else? I guess I don't care that much, just seems like an unnecessary deviation from the norm.